### PR TITLE
chore: Clean up deps and npm audit issues

### PR DESCRIPTION
**Pull request by jdeichert**
## Motivations

This PR cleans up a number of High Severity audit issues (from 19, down to 1). To achieve this, I tried to make the minimum amount of changes required. Along the way, I attempted to clean up some old, unnecessary dependencies as well. Check out my inline comments for more context.

This also closes a number of open Dependabot PRs:
* Closes https://github.com/GetJobber/atlantis/pull/2053
* Closes https://github.com/GetJobber/atlantis/pull/2026
* Closes https://github.com/GetJobber/atlantis/pull/2025
* Closes https://github.com/GetJobber/atlantis/pull/1996
* Closes https://github.com/GetJobber/atlantis/pull/1969
* Closes https://github.com/GetJobber/atlantis/pull/1959

Before |  After
:---:|:---:
Screenshot 2024-10-08 at 11 39 35 AM | Screenshot 2024-10-08 at 11 27 31 AM

### Changed

- Updated plop from v2 to v4
- Updated all storybook deps to match 7.6.20
- Updated react-native to match JM's version
    - Fixed some type-related issues due to this 

### Removed

- Unused dependencies

### Security

- Reduced 19 high sev issues down to 1

## Testing

To test this:

* install the pre-release versions ([comment](https://github.com/GetJobber/atlantis/pull/2058#issuecomment-2400507431)) inside JO/JF/JM
* poke around and make sure everything seems to look normal
    * I ran through creating client->request->quote->job->invoice as a basic flow to ensure things are working nicely
* I also ran through every storybook story to make sure they loaded fine. I didn't notice any problems there.
* Test out plop v4 via `npm run generate` to ensure it still works

Changes can be [tested via Pre-release](https://github.com/GetJobber/atlantis/actions/workflows/trigger-qa-build.yml)

### CODE REVIEW

**Comment by Aiden-Brine** [comment]
Seeing a bug on the home page when I use the pre-release on Jobber Frontend
Screenshot 2024-10-10 at 2 59 51 PM

File: package.json [hunk 1]
```diff
@@ -174,15 +168,21 @@
...
     "node": "18",
     "npm": "10"
   },
   "overrides": {
+    "lerna": {
+      "semver": "^5.7.2 || ^6.3.1 || ^7.5.2",
+      "make-dir": {
+        "semver": "^6.3.1"
+      }
+    },
```
[the remark this answers is not in the document]
Comment (MichaelParadis) [reply package.json:185]: Just a comment, we may want to make a ticket for updating lerna / remove it if we want to transition later but that is a **not in this PR thing**

File: packages/components/package.json [hunk 2]
```diff
@@ -448,10 +448,6 @@
     "@rollup/plugin-commonjs": "^25.0.7",
     "@rollup/plugin-node-resolve": "15.2.3",
     "@rollup/plugin-typescript": "^11.1.6",
-    "@testing-library/jest-dom": "^5.16.5",
```
Comment (MichaelParadis) [inline packages/components/package.json:451]: Changes from the actual bundled output of components is just file renames with different hashes and updates to imports to go the the renamed files

File: packages/design/src/index.ts [hunk 3]
```diff
@@ -111,8 +111,8 @@ function buildSVGStyle(
   const iconSizeStyle = iconSizes.tokens[size];
   const svgStyle: {
     fill?: string;
-    width: string | number;
-    height: string | number;
+    width: number;
```
Comment (MichaelParadis) [inline packages/design/src/index.ts:114]: Checking the compiled output for design it is the exact same except for this but there are no typescript errors caused by this change

File: packages/components-native/src/InputText/InputText.tsx [hunk 4]
```diff
@@ -400,7 +400,6 @@ function InputTextInternal(
           styleOverride?.inputText,
           loading && loadingType === "glimmer" && { color: "transparent" },
         ]}
-        // @ts-expect-error - does exist on 0.71 and up https://github.com/facebook/react-native/pull/39281
         readOnly={readonly}
```
Comment (MichaelParadis) [inline packages/components-native/src/InputText/InputText.tsx:403]: This is the only diff in the bundle output but there are no typescript errors caused by this

**Comment by MichaelParadis** [comment]
@Aiden-Brine I was unable to see this. My assumption is that some file is being caused try cleaning all of node modules and trying a new install.

**Review by MichaelParadis — APPROVED** [review]
LGTM!

**Pull request merged**